src/nplinker/nplinker.py

```diff
@@ -355,3 +356,68 @@ def save_data(
         data = (self.bgcs, self.gcfs, self.spectra, self.mfs, self.strains, links)
         with open(file, "wb") as f:
             pickle.dump(data, f)
+
+    def objects_to_tsv(self, objects: Sequence[BGC] | Sequence[Spectrum], filename: str) -> None:
+        """Exports a list of BGC or Spectrum objects to a tsv file.
+
+        Args:
+            objects (list): A list of BGC or a list of Spectrum objects to be exported.
+            filename (str): The name of the output file.
+        """
+        if not objects:
+            raise ValueError("No objects provided to export")
+
+        # Ensure all elements in the list are of the same type
+        obj_type = type(objects[0])
+        if not all(isinstance(obj, obj_type) for obj in objects):
+            raise TypeError("All objects in the list must be of the same type")
+
+        with open(self._output_dir / filename, "w", newline="") as outfile:
+            headers = objects[0].to_dict().keys()
+            writer = csv.DictWriter(outfile, fieldnames=headers, delimiter="\t")
+            writer.writeheader()
+            for obj in objects:
+                row = obj.to_dict()
+                for key, value in row.items():
+                    row[key] = self.to_string(value).replace("\t", "    ")
+                writer.writerow(row)
```

Comment:
```suggestion
            for obj in objects:
                writer.writerow(obj.to_tabular())
```

It's better to implement the transformation in each object class, i.e. `to_tabluar` (or your `to_string`), which is more modular. So that wherever it's needed, you just call it instead of implementing it again.